device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/device/Device.kt

```diff
@@ -366,9 +366,26 @@ class Device(
         remote.fbsimctl.uninstallApp(udid, bundleId)
     }
 
+    fun runXcuiTest(xcuiTestExecutionConfig: XcuiTestExecutionConfig): Map<String, String> {
+        logger.debug(logMarker, "Running XCUI test '${xcuiTestExecutionConfig.testName}' of '${xcuiTestExecutionConfig.appName}'" +
+                " using '${xcuiTestExecutionConfig.pathToDirWithXctestrunFile}/${xcuiTestExecutionConfig.xctestrunFileName}' on device $this")
+
+        val command = listOf("xcodebuild", "test-without-building", "-xctestrun",
+                "${xcuiTestExecutionConfig.pathToDirWithXctestrunFile}/${xcuiTestExecutionConfig.xctestrunFileName}",
+                "-destination", "platform=iOS,id=$udid", "-only-testing:${xcuiTestExecutionConfig.testName}")
+
+        val result = remote.execIgnoringErrors(command, timeOutSeconds = xcuiTestExecutionConfig.timeoutSec)
+        return mapOf(
+                "command" to result.cmd.joinToString(" "),
```

Comment:
Could you please create a class XcuiTestExecutionResult and use it here instead of Map<String, String>

Reply:
Done.
#36